lib/shared/Button/button_widget.dart

```diff
@@ -0,0 +1,158 @@
+  List<Color> stylePurple = [BSColors().purpleButton.darken(0.1), BSColors().purpleButton, BSColors().purpleButton.brighten(0.4), BSColors().purpleButton.brighten(0.2), BSColors().purpleDarkBorder, BSColors().whiteText];
+  List<Color> styleGray = [BSColors().darkGreyButton.darken(0.1), BSColors().darkGreyButton, BSColors().darkGreyButton.brighten(0.4), BSColors().darkGreyButton.brighten(0.2), BSColors().blackBorder, BSColors().whiteText];
+  List<Color> styleFb = [BSColors().blueButton.darken(0.1), BSColors().blueButton, BSColors().blueButton.brighten(0.4), BSColors().blueButton.brighten(0.2), BSColors().fbBorder, BSColors().whiteText];
+  List<Color> styleGoogle = [BSColors().redButton.darken(0.1), BSColors().redButton, BSColors().redButton.brighten(0.4), BSColors().redButton.brighten(0.2), BSColors().googleBorder, BSColors().whiteText];
+  List<Color> styleApple = [BSColors().greyButton.darken(0.1), BSColors().greyButton, BSColors().greyButton.brighten(0.4), BSColors().greyButton.brighten(0.2), BSColors().blackBorder, BSColors().whiteText];
+
+  setStyle(ButtonType type, number) {
+        switch (type) {
+          case ButtonType.orange:
+            return styleOrange[number];
+          case ButtonType.purple:
+            return stylePurple[number];
+          case ButtonType.gray:
+            return styleGray[number];
+          case ButtonType.fb:
+            return styleFb[number];
+          case ButtonType.google:
+            return styleGoogle[number];
+          case ButtonType.apple:
+            return styleApple[number];
+        }
+  }
+
+  setContent(ContentType type) {
+    switch (type) {
+      case ContentType.icon:
+        return const Icon(Icons.abc);
+      case ContentType.text:
+        return Text(
+    widget.title.toUpperCase(),
+    textAlign: TextAlign.center,
+    overflow: TextOverflow.visible,
+    maxLines: 1,
+    style: TextStyle(
+    fontFamily: BattleShipFont.policy.titleLarge.toString(),
+    fontWeight: FontWeight.bold,
+    color: setStyle(widget.buttonType, 5),
+    fontSize: 30,
+    ),
+    );
+      case ContentType.textIcon:
+        return Row(
+          children: [
+            const Icon(Icons.abc),
+            Text(
+              widget.title.toUpperCase(),
+              textAlign: TextAlign.center,
+              overflow: TextOverflow.visible,
+              maxLines: 1,
+              style: TextStyle(
+                fontFamily: BattleShipFont.policy.titleLarge.toString(),
+                fontWeight: FontWeight.bold,
+                color: setStyle(widget.buttonType, 5),
+                fontSize: 30,
+              ),
+            ),
+          ],
+        );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      onTap: () {
+        print('c\'est un bouton');
```

Comment:
Does it works ? _Don't change anything, I am just curious_
```suggestion
        print("c'est un bouton");
```

Reply:
Just for testing if the button is clicked, it displays a message in the console, no use even so